content/performance/docs/binpack.md

```diff
@@ -0,0 +1,421 @@
+# **Custom Scheduler for Binpacking**
+Binpacking directly influences the cost performance of your EKS cluster. Both Cluster Autoscaler and Karpenter, as node provisioners, can efficiently scales nodes in and out, but without a binpacking custom scheduler, you might end up with a scenario where nodes are underutilized due to inefficient pod distribution at the pod scheduling time.
+
+-   **Better Binpacking:** When pods are tightly packed onto fewer nodes, node provisioner (Cluster Autoscaler or Karpenter) can more effectively scale down unused nodes, reducing costs.
+-   **Poor Binpacking:** If pods are spread out inefficiently across many nodes, node provisioner might struggle to find opportunities to scale down, leading to resource waste.
+
+## Demonstration
+**Poor Binpacking with the default kube-scheduler:**
+<p align="center">
+  <img src="../resources/images/nonbinpack.gif" width="640" height="400"/>
+</p>
+
+**Better Binpacking with a Custom Scheduler:**
+<p align="center">
+  <img src="../resources/images/binpack.gif" width="640" height="400"/>
+</p>
+
+Before we tackle the binpacking issue in Amazon EMR on EKS, let’s install a node viewer tool - eks-node-viewer. It 
+provides enhanced visibility into the EKS nodes utilization, helping with real-time monitoring, debugging, and optimization efforts. In this case, we use it to visualize dynamic node and pod allocation within an EKS cluster for tracking the binpacking performance.
+
+## Install eks-node-viewer
+
+eks-node-viewer：https://github.com/awslabs/eks-node-viewer
+
+**HomeBrew**
+```bash
+brew tap aws/tap
+brew install eks-node-viewer
+```
+**Manual**
+```bash
+go install github.com/awslabs/eks-node-viewer/cmd/eks-node-viewer@latest
+```
+
+Quick Start:
+```bash
+eks-node-viewer --resources cpu,memory
+```
+
+## Install Bin-packing custom scheduler
+
+In the [scheduling-plugin](https://kubernetes.io/docs/reference/scheduling/config/#scheduling-plugins) ` NodeResourcesFit` of kube-scheduler, there are two scoring strategies that support the bin packing of resources:       `MostAllocated` and `RequestedToCapacityRatio`. We created a custom scheduler based on the **MostAllocated strategy**. See the K8s’s [Resource Bin Packing](https://kubernetes.io/docs/concepts/scheduling-eviction/resource-bin-packing/) documentation for more details.
+
+The following customer scheduler named “my-scheduler” is created for EKS version v1.28, as the “kube-scheduler” container image version and “KubeSchedulerConfiguration“ API version are set for the v1.28. Please adjust them accordingly, if your EKS version is different.  
```

Comment:
I think we need to be more explicit that a user need to use an image/api version that match the k8s they are running.

Reply:
see the [improve binpack](https://github.com/aws/aws-emr-containers-best-practices/pull/57/commits/55123ad158cd508fbc36ab7faedba6ef833598d4)